### DISCUSSION TITLE
Github Action fails - conanfile.py missing build_test

### DISCUSSION THREAD

**Discussion by robot-ranger** [mtconnect/cppagent#296] [Q&A]
I dont know enough about github actions, so i am not sure this is a bug or user error.

building docker image seems to fail at the make installer step:

` #11 [build 4/5] RUN conan export conan/mqtt_cpp   && conan install . -if build --build=missing   -pr conan/profiles/docker   -o build_tests=False   -o run_tests=False
#11 3.111 WARN: Remotes registry file missing, creating default one in /root/.conan/remotes.json
#11 3.134 Exporting package recipe
#11 3.147 mqtt_cpp/13.1.0: A new conanfile.py version was exported
#11 3.147 mqtt_cpp/13.1.0: Folder: /root/.conan/data/mqtt_cpp/13.1.0/_/_/export
#11 3.151 mqtt_cpp/13.1.0: Exported revision: 745bd468b7493340b0c9af2b50fec6ee
#11 6.287 Auto detecting your dev setup to initialize the default profile (/root/.conan/profiles/default)
#11 6.736 Found gcc 11
#11 6.773 gcc>=5, using the major as version
#11 6.776 
#11 6.776 ************************* WARNING: GCC OLD ABI COMPATIBILITY ***********************
#11 6.776  
#11 6.776 Conan detected a GCC version > 5 but has adjusted the 'compiler.libcxx' setting to
#11 6.776 'libstdc++' for backwards compatibility.
#11 6.776 Your compiler is likely using the new CXX11 ABI by default (libstdc++11).
#11 6.776 
#11 6.776 If you want Conan to use the new ABI for the default profile, run:
#11 6.776 
#11 6.776     $ conan profile update settings.compiler.libcxx=libstdc++11 default
#11 6.776 
#11 6.776 Or edit '/root/.conan/profiles/default' and set compiler.libcxx=libstdc++11
#11 6.776 
#11 6.776 ************************************************************************************
#11 6.776 
#11 6.776 
#11 6.776 
#11 6.776 Default settings
#11 6.777 	os=Linux
#11 6.777 	os_build=Linux
#11 6.777 	arch=armv8
#11 6.777 	arch_build=armv8
#11 6.777 	compiler=gcc
#11 6.777 	compiler.version=11
#11 6.777 	compiler.libcxx=libstdc++
#11 6.777 	build_type=Release
#11 6.778 *** You can change them in /root/.conan/profiles/default ***
#11 6.778 *** Or override with -s compiler='other' -s ...s***
#11 6.778 
#11 6.778 
#11 7.182 Configuration:
#11 7.183 [settings]
#11 7.183 arch=armv8
#11 7.183 arch_build=armv8
#11 7.183 build_type=Release
#11 7.183 compiler=gcc
#11 7.183 compiler.libcxx=libstdc++11
#11 7.183 compiler.version=11
#11 7.183 os=Linux
#11 7.183 os_build=Linux
#11 7.183 [options]
#11 7.183 build_tests=False
#11 7.183 run_tests=False
#11 7.183 without_ipv6=True
#11 7.183 [build_requires]
#11 7.183 [env]
#11 7.183 
#11 7.217 ERROR: /root/agent/conanfile.py: option 'build_tests' doesn't exist
#11 7.217 Possible options are ['without_ipv6', 'with_ruby', 'development', 'shared', 'winver', 'with_docs', 'cpack', 'agent_prefix', 'fPIC']
#11 ERROR: process "/bin/sh -c conan export conan/mqtt_cpp   && conan install . -if build --build=missing   -pr $CONAN_PROFILE   -o build_tests=$WITH_TESTS   -o run_tests=$WITH_TESTS" did not complete successfully: exit code: 1
------
 > [build 4/5] RUN conan export conan/mqtt_cpp   && conan install . -if build --build=missing   -pr conan/profiles/docker   -o build_tests=False   -o run_tests=False:
7.183 os_build=Linux
7.183 [options]
7.183 build_tests=False
7.183 run_tests=False
7.183 without_ipv6=True
7.183 [build_requires]
7.183 [env]
7.183 
7.217 ERROR: /root/agent/conanfile.py: option 'build_tests' doesn't exist
7.217 Possible options are ['without_ipv6', 'with_ruby', 'development', 'shared', 'winver', 'with_docs', 'cpack', 'agent_prefix', 'fPIC']
------
Dockerfile:65
--------------------
  64 |     # make installer
  65 | >>> RUN conan export conan/mqtt_cpp \
  66 | >>> #   && conan export conan/mruby \
  67 | >>>   && conan install . -if build --build=missing \
  68 | >>>   -pr $CONAN_PROFILE \
  69 | >>>   -o build_tests=$WITH_TESTS \
  70 | >>>   -o run_tests=$WITH_TESTS
  71 |     #   -o with_ruby=$WITH_RUBY
--------------------
ERROR: failed to solve: process "/bin/sh -c conan export conan/mqtt_cpp   && conan install . -if build --build=missing   -pr $CONAN_PROFILE   -o build_tests=$WITH_TESTS   -o run_tests=$WITH_TESTS" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c conan export conan/mqtt_cpp   && conan install . -if build --build=missing   -pr $CONAN_PROFILE   -o build_tests=$WITH_TESTS   -o run_tests=$WITH_TESTS" did not complete successfully: exit code: 1`

id be happy to make this an issue, but thought i would see if any experts can spot something im doing or not doing first?

**Comment by robot-ranger** [comment]
sorry for the formatting.

as you can see - i tried removing ruby as that has helped in the past, but it didnt help here

**Comment by wsobel** [comment]
I’ll take a look at the docker build. Just finished the other builds and got actions working. The changes were more than I had originally planned because I needed to convert to the new package structure. I’ll check it out and see what’s going wrong. I’ll them add the docker build to CI/CD. BestW(Sent from mobile)On Jul 26, 2023, at 13:25, robot-ranger ***@***.***> wrote:﻿
sorry for the formatting.
as you can see - i tried removing ruby as that has helped in the past, but it didnt help here

**Comment by robot-ranger** [comment]
ive added `"build_tests"` and `"run_tests"` options to the conanfile.py and that gets me into the `conan build` part, but new errors pop up:

`
#19 [linux/amd64 build 5/5] RUN conan build . -bf build
#19 1.392 Using lockfile: '/root/agent/build/conan.lock'
#19 1.392 Using cached profile from lockfile
#19 1.875 Exporting package recipe
#19 1.879 mqtt_cpp/13.1.0: The stored package has not changed
#19 1.880 mqtt_cpp/13.1.0: Exported revision: 745bd468b7493340b0c9af2b50fec6ee
#19 1.956 conanfile.py (mtconnect_agent/2.2): Calling build()
#19 1.959 ERROR: conanfile.py (mtconnect_agent/2.2): Error in build() method, line 157
#19 1.959 	cmake = CMake(self)
#19 1.959 	ConanException: CMakePresets.json was not found in /root/agent/build/build/Release/generators folder. Check that you are using CMakeToolchain as generator to ensure its correct initialization.
#19 ERROR: process "/bin/sh -c conan build . -bf build" did not complete successfully: exit code: 1

#18 [linux/arm64 build 4/5] RUN conan export conan/mqtt_cpp   && conan export conan/mruby   && conan install . -if build --build=missing   -pr conan/profiles/docker   -o build_tests=False   -o run_tests=False   -o with_ruby=False
#18 CANCELED
------
 > [linux/amd64 build 5/5] RUN conan build . -bf build:
1.392 Using lockfile: '/root/agent/build/conan.lock'
1.392 Using cached profile from lockfile
1.875 Exporting package recipe
1.879 mqtt_cpp/13.1.0: The stored package has not changed
1.880 mqtt_cpp/13.1.0: Exported revision: 745bd468b7493340b0c9af2b50fec6ee
1.956 conanfile.py (mtconnect_agent/2.2): Calling build()
1.959 ERROR: conanfile.py (mtconnect_agent/2.2): Error in build() method, line 157
1.959 	cmake = CMake(self)
1.959 	ConanException: CMakePresets.json was not found in /root/agent/build/build/Release/generators folder. Check that you are using CMakeToolchain as generator to ensure its correct initialization.
------
Dockerfile:74
--------------------
  72 |     
  73 |     # compile source (~20mins - 4hrs for qemu)
  74 | >>> RUN conan build . -bf build
  75 |     
  76 |     # ---------------------------------------------------------------------
--------------------
ERROR: failed to solve: process "/bin/sh -c conan build . -bf build" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c conan build . -bf build" did not complete successfully: exit code: 1`

**Comment by wsobel** [comment]
Conan 2 has a different way of managing packages and tests. I will modify the docker build for the new syntax. (Sent from mobile)On Jul 26, 2023, at 14:06, robot-ranger ***@***.***> wrote:﻿
ive added "build_tests" and "run_tests" options to the conanfile.py and that gets me into the conan build part, but new errors pop up:
`
#19 [linux/amd64 build 5/5] RUN conan build . -bf build
#19 1.392 Using lockfile: '/root/agent/build/conan.lock'
#19 1.392 Using cached profile from lockfile
#19 1.875 Exporting package recipe
#19 1.879 mqtt_cpp/13.1.0: The stored package has not changed
#19 1.880 mqtt_cpp/13.1.0: Exported revision: 745bd468b7493340b0c9af2b50fec6ee
#19 1.956 conanfile.py (mtconnect_agent/2.2): Calling build()
#19 1.959 ERROR: conanfile.py (mtconnect_agent/2.2): Error in build() method, line 157
#19 1.959 	cmake = CMake(self)
#19 1.959 	ConanException: CMakePresets.json was not found in /root/agent/build/build/Release/generators folder. Check that you are using CMakeToolchain as generator to ensure its correct initialization.
#19 ERROR: process "/bin/sh -c conan build . -bf build" did not complete successfully: exit code: 1
#18 [linux/arm64 build 4/5] RUN conan export conan/mqtt_cpp   && conan export conan/mruby   && conan install . -if build --build=missing   -pr conan/profiles/docker   -o build_tests=False   -o run_tests=False   -o with_ruby=False
#18 CANCELED

[linux/amd64 build 5/5] RUN conan build . -bf build:
1.392 Using lockfile: '/root/agent/build/conan.lock'
1.392 Using cached profile from lockfile
1.875 Exporting package recipe
1.879 mqtt_cpp/13.1.0: The stored package has not changed
1.880 mqtt_cpp/13.1.0: Exported revision: 745bd468b7493340b0c9af2b50fec6ee
1.956 conanfile.py (mtconnect_agent/2.2): Calling build()
1.959 ERROR: conanfile.py (mtconnect_agent/2.2): Error in build() method, line 157
1.959 	cmake = CMake(self)
1.959 	ConanException: CMakePresets.json was not found in /root/agent/build/build/Release/generators folder. Check that you are using CMakeToolchain as generator to ensure its correct initialization.


Dockerfile:74
72 |
73 |     # compile source (~20mins - 4hrs for qemu)
74 | >>> RUN conan build . -bf build
75 |
76 |     # ---------------------------------------------------------------------
ERROR: failed to solve: process "/bin/sh -c conan build . -bf build" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c conan build . -bf build" did not complete successfully: exit code: 1`

**Reply by robot-ranger** [reply]
the dockerfile has had `&& pip install conan -v "conan==1.59.0"`

since https://github.com/mtconnect/cppagent/commit/bffb7b347e74d2c743818cc52fce71e4c2564ec9

...are we using conan 2?

**Comment by wsobel** [comment]
Yes. The new build will only work with 2.0. I think the current version is 2.0.8(Sent from mobile)On Jul 26, 2023, at 15:43, robot-ranger ***@***.***> wrote:﻿
the dockerfile has had && pip install conan -v "conan==1.59.0"
since bffb7b3
...are we using conan 2?

**Comment by robot-ranger** [comment]
looks like docker image builds good in github actions! 👍 thx for the edits a couple days ago!
